cli/src/main/java/com/devonfw/tools/ide/io/IniFileImpl.java

```diff
@@ -0,0 +1,61 @@
+package com.devonfw.tools.ide.io;
+
+import java.util.LinkedHashMap;
+
+/**
+ * Implementation of {@link IniFile} preserves order of sections and properties between reading and writing
+ */
+public class IniFileImpl implements IniFile {
+
+  LinkedHashMap<String, IniSection> iniMap;
+
+  /**
+   * creates empty IniFileImpl
+   */
+  public IniFileImpl() {
+    this.iniMap = new LinkedHashMap<>();
+  }
+
+  @Override
+  public String[] getSectionNames() {
+    return iniMap.keySet().toArray(new String[0]);
+  }
+
+  @Override
+  public boolean removeSection(String section) {
+    boolean sectionExists = iniMap.containsKey(section);
+    iniMap.remove(section);
+    return sectionExists;
+  }
+
+  @Override
+  public IniSection getSection(String section) {
+    return iniMap.get(section);
+  }
+
+  @Override
+  public IniSection getOrCreateSection(String section) {
+    IniSection iniSection;
+    if (!iniMap.containsKey(section)) {
+      iniSection = new IniSectionImpl(section);
+      iniMap.put(section, iniSection);
+    } else {
+      iniSection = iniMap.get(section);
+    }
+    return iniSection;
```

Comment:
Can be simplified like this:
```suggestion
    return this.iniMap.computeIfAbsent(section, IniSectionImpl::new);
```